source/arriving/setup_computer.rst

```diff
@@ -1,51 +1,167 @@
+
+.. code-block:: bash
+
+        sudo add-apt-repository ppa:deadsnakes/ppa
+        sudo apt-get install python3.10-dev python3.10-venv python3.10-tk python3-pip
+        python3.10 -m pip install pip  # update pip (python package manager) to latest version
+
+Other python packages can be installed by replacing python3.X by a version of your choice.
+
+Virtual environments
+""""""""""""""""""""
+
+Virtual environments are a good way to separate the system's python installation from the python version you need for your work. It also allows you to have a precise set of python packages with specific versions. While there are many options for environments, we highly recommand using `VirtualEnv <https://virtualenv.pypa.io/en/latest/>`_. Start by installing it and creating a directory where all your environments will be saved:
+
+.. code-block:: bash
+
+    sudo pip install virtualenvwrapper
+    mkdir ~/Envs
+
+Then you should add these lines in your .bashrc (:bash:`code ~/.bashrc`):
+
+.. code-block:: bash
+
+    export WORKON_HOME=~/Envs/
+    #export VIRTUALENVWRAPPER_PYTHON=/usr/bin/python3   # uncomment this line if you have errors when starting your terminal (next step)
+    source /usr/local/bin/virtualenvwrapper.sh
+
+Finally, you can restart the terminal and create an environment:
+
+.. code-block:: bash
+
+    NAME=somename # Ex: $NAME could be my_env_py3
+                    # depending on the use you will give to this environment.
+                    # Try to be explicit
+    # One of:
+    mkvirtualenv $NAME
+    mkvirtualenv --python=python3.10 $NAME   # To use one precise python version
+                                                # You can also use the complete path to the python installation
+    mkvirtualenv $NAME --python=/usr/bin/python3.10  # Equivalent. The python
+                                                        # version must be already
+                                                        # installed on your computer
+
+    # Note. To download a python version to your /usr/bin without installing it:
+    # https://www.python.org/downloads/
+
+When everything is set, restart your terminal. You should now be able to work in a chosen environmnent by using :bash:`workon $NAME`. Now everytime you need to install a new library inside the environment, you should always try to use :bash:`pip install your_library`. The environment will use the right pip based on the current python version. *Don't use sudo pip!* It will use the pip of the system! If the library can't be installed via pip, however, you can use :bash:`sudo apt-get install your_library`. In this case the `sudo` will not be a problem.
```

Comment:
That last line make it seem like sudo apt-get will safely install any lib into your venv no? I think I would leave that last sentence out.

Reply:
environmnent -> environment